playbooks/files/library/sr_fingerprint.py

```
@@ -51,33 +170,187 @@ def _local_iso8601_no_microseconds():
-        _local_iso8601_no_microseconds(),
-    )
+def _format_fingerprint_jsonl(record):
+    """Format the canonical fingerprint record as a single JSON line."""
+    return json.dumps(record, separators=(",", ":"), sort_keys=False)
+
+
+def _trim_log_file(log_file, target_size):
+    """Remove oldest records until the file fits in target_size bytes."""
+    with open(log_file, "r") as log_fd:
+        lines = log_fd.readlines()
+    while lines and sum(len(l) for l in lines) > target_size:
```

Ruff (0.16.1)

[error] Ambiguous variable name: `l`

(E741)